[eigen3] Build failure on arm64-osx

Package: eigen3:arm64-osx@3.4.1

**Host Environment**

- Host: arm64-osx
- Compiler: AppleClang 17.0.0.17000319
-    vcpkg-tool version: 2025-07-21-d4b65a2b83ae6c3526acd1c6f3b51aff2a884533
    vcpkg-scripts version: 0d9d468435 2025-09-30 (2 days ago)

**To Reproduce**

`vcpkg install eigen3`

**Failure logs**

```
Downloading https://gitlab.com/libeigen/eigen/-/archive/3.4.1/eigen-3.4.1.tar.gz -> libeigen-eigen-3.4.1.tar.gz
libeigen-eigen-3.4.1.tar.gz.14884.part: error: download from https://gitlab.com/libeigen/eigen/-/archive/3.4.1/eigen-3.4.1.tar.gz had an unexpected hash
note: Expected: 8b82e6785eda1982fd8f2a8321b0fff144062eb78cac6b9d0c87cba374cfaa3f2bde33a34069b2871ae548ce681f5a7e1247918c5215d92ed4836614548a1936
note: Actual  : 15b540d2fad25de837d7612456ea65b7fcfa25f886c48dcb3de025e079f5f43f45f1c0bd9a8435430d0e2fb9f0a6129719626dd4fd45dc61fc480f97762b2f56
CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:136 (message):
  Download failed, halting portfile.
Call Stack (most recent call first):
  scripts/cmake/vcpkg_from_gitlab.cmake:113 (vcpkg_download_distfile)
  ports/eigen3/portfile.cmake:3 (vcpkg_from_gitlab)
  scripts/ports.cmake:206 (include)



```

--- a/ports/eigen3/vcpkg.json
+++ b/ports/eigen3/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "eigen3",
   "version": "3.4.1",
+  "port-version": 1,
   "description": "C++ template library for linear algebra: matrices, vectors, numerical solvers, and related algorithms.",
   "homepage": "http://eigen.tuxfamily.org",
   "license": "MPL-2.0",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2678,7 +2678,7 @@
     },
     "eigen3": {
       "baseline": "3.4.1",
-      "port-version": 0
+      "port-version": 1
     },
     "eipscanner": {
       "baseline": "1.3.0",

--- a/versions/e-/eigen3.json
+++ b/versions/e-/eigen3.json
@@ -1,7 +1,12 @@
 {
   "versions": [
     {
-      "git-tree": "e4983ed273755fd9e95b45abe264eb028db11724",
+      "git-tree": "ea6ed04401cca325eff95c17cc0d01326a3c6a85",
+      "version": "3.4.1",
+      "port-version": 1
+    },
+    {
+      "git-tree": "86e81b275f5e8d1643d5204b1f5edaf7bfcbde5b",
       "version": "3.4.1",
       "port-version": 0
     },

